File: seed/csharp-sdk/exhaustive/serialize-null-properties/src/SeedExhaustive/Types/Object/Types/NestedObjectWithOptionalField.cs
using System.Text.Json.Serialization;
using SeedExhaustive.Core;

#nullable enable

namespace SeedExhaustive.Types;

public record NestedObjectWithOptionalField
{
    [JsonPropertyName("string")]
    public string? String { get; set; }

    [JsonPropertyName("NestedObject")]
    public ObjectWithOptionalField? NestedObject { get; set; }

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/ruby-sdk-v2/exhaustive/wire-tests/lib/fern_exhaustive/types/object/types/nested_object_with_optional_field.rb
# frozen_string_literal: true

module FernExhaustive
  module Types
    module Object_
      module Types
        class NestedObjectWithOptionalField < Internal::Types::Model
          field :string, -> { String }, optional: true, nullable: false
          field :nested_object, -> { FernExhaustive::Types::Object_::Types::ObjectWithOptionalField }, optional: true, nullable: false, api_name: "NestedObject"
        end
      end
    end
  end
end
